pkg/services/alerting/engine.go

```diff
@@ -77,7 +77,7 @@ func ProvideAlertEngine(renderer rendering.Service, requestValidator models.Plug
 		sqlStore:           sqlStore,
 		dashAlertExtractor: dashAlertExtractor,
 	}
-	e.ticker = NewTicker(time.Now(), time.Second*0, clock.New(), 1)
+	e.ticker = NewTicker(clock.New(), 1*time.Second)
```

Comment:
I suppose if we use `context.Context` for `NewTicker` then a stopped ticker cannot be restarted, where as if we use it in `run` then a stopped ticker could be restarted.

Reply:
I am not sure that ticker should have a `context.Context` as its more like a background process that should be started and stopped rather than cancelled. I think this makes sense as `time.Ticker` and `time.After` cannot be canceled either.

Reply:
I am not sure that ticker should have a context.Context as its more like a background process that should be started and stopped rather than cancelled. I think this makes sense as `time.Ticker` and `time.Timer` cannot be canceled either.

Reply:
Correct me if I'm wrong, but if Grafana it's shutting down should't the ticker be also cancelled?

Reply:
Yes, but we can use `ticker.Stop()`. I might be incorrect here but I would argue its idiomatic to use `context.Context` for per-request (doesn't have to be an HTTP request) cancelation while `Close()` and `Stop()` should be used for stopping background processes (i.e. `ticker.Stop()` in `time` package and `server.Close()` in `net/http` package).

Reply:
Yes but we can use `ticker.Stop()`. I might be incorrect here but I would argue its idiomatic to use `context.Context` for per-request (request here doesn't have to be a `net/http` request) cancelation while `Close()` and `Stop()` should be used for stopping background processes (i.e. `ticker.Stop()` in `time` package and `server.Close()` in `net/http` package).

Reply:
Great, thanks for the suggestion! I'm happy to go for the `ticker.Stop()` approach.